core/azure-core/src/test/java/com/azure/core/polling/PollerTests.java

```diff
@@ -0,0 +1,375 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.polling;
+
+import com.azure.core.exception.HttpResponseException;
+/*import com.azure.core.polling.PollResponse.OperationStatus;
+
+import org.junit.Assert;
+import org.junit.Test;
+import reactor.test.StepVerifier;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;*/
+import java.time.LocalDateTime;
+import java.util.Collection;
+import java.util.Date;
+import java.util.function.Function;
+
+public class PollerTests {
+
+    private boolean debug = true;
+
+    private Function<PollResponse<CreateCertificateResponse>, PollResponse<CreateCertificateResponse>> createPollOperation(
+        PollResponse<CreateCertificateResponse> intermediateProgressPollResponse,
+        PollResponse<CreateCertificateResponse> finalPollResponse,
+        int pollIntervalInMillis,
+        int sendFinalResponseInSeconds
+    ) {
+        return new Function<PollResponse<CreateCertificateResponse>, PollResponse<CreateCertificateResponse>>() {
+            // Will return success after this time.
+            LocalDateTime timeToReturnSuccess = LocalDateTime.now().plusSeconds(sendFinalResponseInSeconds);
+
+            @Override
+            public PollResponse<CreateCertificateResponse> apply(PollResponse<CreateCertificateResponse> prePollResponse) {
+                if (LocalDateTime.now().isBefore(timeToReturnSuccess)) {
+                    debug(" Service poll function called ", " returning intermediate response ");
+                    return intermediateProgressPollResponse;
+                } else {
+                    debug("   Service poll function called ", " returning final response ");
+                    return finalPollResponse;
+                }
+            }
+        };
+    }
+
+    /* Test where SDK Client is subscribed all responses.
+     * The last response in this case will be PollResponse.OperationStatus.SUCCESSFULLY_COMPLETED
+     * This scenario is setup where source will generate successful response returned after few in-progress response.
+     **/
+    /*@Test
```

Comment:
To go along with the import question, is everything commented out for the time being?

Reply:
yes, please ignore unit test for time being